projects/ad7746-ebz/src/app/headless.c

```diff
@@ -0,0 +1,170 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <sys/platform.h>
+#include "adi_initialize.h"
+#include <drivers/pwr/adi_pwr.h>
+
+#include "ad7746.h"
+#include "i2c.h"
+#include "delay.h"
+#include "print_log.h"
+#include "uart.h"
+#include "uart_extra.h"
+#include "uart_stdio.h"
+
+
+int32_t platform_init()
+{
+	if (ADI_PWR_SUCCESS != adi_pwr_Init())
+		return FAILURE;
+
+	if (ADI_PWR_SUCCESS != adi_pwr_SetClockDivider(ADI_CLOCK_HCLK, 1u))
+		return FAILURE;
+
+	if (ADI_PWR_SUCCESS != adi_pwr_SetClockDivider(ADI_CLOCK_PCLK, 1u))
+		return FAILURE;
+
+	adi_initComponents();
+	return 0;
+}
+
+int main(void)
+{
+	int32_t ret;
+	char input[10];
+	uint32_t capData = 0;
+	uint32_t min = 0;
+	uint32_t max = 0;
+	uint32_t mm = 0;
+	uint32_t temperature = 0;
+	struct ad7746_init_param adcip;
+	struct ad7746_dev *adc = NULL;
+
+	struct uart_desc *uart;
+	struct aducm_uart_init_param xuip = {
+		.parity = UART_NO_PARITY,
+		.stop_bits = UART_ONE_STOPBIT,
+		.word_length = UART_WORDLEN_8BITS
+	};
+	struct uart_init_param uip = {
+		.device_id = 0,
+		.baud_rate = 115200,
+		.size = UART_CS_8,
+		.parity = UART_PAR_NO,
+		.stop = UART_STOP_1,
+		.extra = &xuip,
+	};
+	struct i2c_platform_ops ops = {
+		.i2c_ops_init = i2c_init,
+		.i2c_ops_remove = i2c_remove,
+		.i2c_ops_write = i2c_write,
+		.i2c_ops_read = i2c_read,
+	};
+
+	ret = platform_init();
+	if (ret < 0)
+		goto error;
+	ret = uart_init(&uart, &uip);
+	if (ret < 0)
+		goto error;
+
+	init_uart_stdio(uart);
+
+	adcip.i2c_init = (struct i2c_init_param) {
+		.max_speed_hz = 100000,
```

Comment:
Can this be a define?. Maybe in parameters.h

Reply:
Same for uart parameters

Reply:
done